ext/fg/js/frontend-api-receiver.js

```diff
@@ -35,14 +35,29 @@ class FrontendApiReceiver {
     _onMessage(port, {id, action, params, target, senderId}) {
         if (target !== this._source) { return; }
 
-        const handler = this._handlers.get(action);
-        if (typeof handler !== 'function') { return; }
+        const messageHandler = this._messageHandlers.get(action);
+        if (typeof messageHandler === 'undefined') { return; }
```

Comment:
An undefined check is used for other message handlers which are objects. The rationale being that `Map.get` returns undefined for things which don't exist, so it's the most simple check. Types other than object would be a developer error, and `isObject` would suppress that.

Reply:
I see, it could indeed be hard to debug if it silently failed for the wrong kind of data. Alternatively this could also throw for any non-object, but JS already will if you try to call `undefined` after destructuring an array for example.